File: 백준/Silver/6588. 골드바흐의 추측/골드바흐의 추측.py
import sys
input = sys.stdin.readline

def getPrimeArr(num):
    primes = [True] * (num + 1)

    for i in range(3, int(num ** 0.5) + 1, 2):
        if primes[i]:
            for j in range(i * i, num + 1, 2 * i):
                primes[j] = False

    return primes

def findFactor(num, primes):

    for i in range(3, num, 2):
        if primes[i] and primes[num - i]:
            return "{0} = {1} + {2}".format(num, i, num - i)

    return "Goldbach's conjecture is wrong."

primes = getPrimeArr(1000001)
while True:
    num = int(input().rstrip())
    
    if num == 0:
        break
    else:
        print(findFactor(num, primes))

File: 백준/Silver/6588. 골드바흐의 추측/골드바흐의 추측.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.util.Arrays;

class Main {

    static boolean[] getPrimeArr(int n) {
        boolean[] primes = new boolean[n + 1];
        Arrays.fill(primes, true);

        for (int i = 3; i <= Math.sqrt(n); i += 2) {
            if (primes[i]) {
                for (int j = i * i; j <= n; j += 2 * i) {
                    primes[j] = false;
                }
            }
        }
        return primes;
    }

    static String findFactor(int input, boolean[] primes) {

        for (int i = 3; i <= input / 2; i += 2) {
            if (primes[input - i] && primes[i]) {
                return input + " = " + i + " + " + (input - i);
            }
        }

        return "Goldbach's conjecture is wrong.";
    }

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

        boolean[] primes = getPrimeArr(1000001);

        while (true) {
            int input = Integer.parseInt(br.readLine());
            if (input == 0) {
                break;
            } else {
                bw.write(findFactor(input, primes) + "\n");
            }
        }
        bw.flush();
        bw.close();
    }
}